explorer/e2e/anvil/anvil-filters.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { expect, test } from "@playwright/test";
+import {
+  filterRegex,
+  getFirstElementTextLocator,
+  testClearAll,
+  testFilterBubbles,
+  testFilterCounts,
+  testFilterPersistence,
+  testFilterPresence,
+} from "../testFunctions";
+import { anvilFilters, anvilTabs, anvilTabTestOrder } from "./anvil-tabs";
+
+const filter_index_list = [3, 4, 5, 10, 6, 2];
+const filter_index_list_short = [1, 10, 3];
+test("Check that all filters exist on the Datasets tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.datasets, anvilFilters);
+});
+
+test("Check that all filters exist on the Donors tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.donors, anvilFilters);
+});
+
+test("Check that all filters exist on the BioSamples tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.biosamples, anvilFilters);
+});
+
+test("Check that all filters exist on the Activities tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.activities, anvilFilters);
+});
+
+test("Check that all filters exist on the Files tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.files, anvilFilters);
+});
+
+test("Check that the first filter on the Datasets tab creates at least one checkbox, and that checking up to the first five does not cause an error and does not cause there to be no entries in the table", async ({
+  page,
+}) => {
+  // Goto the datasets tab
+  await page.goto(anvilTabs.datasets.url);
+  await expect(
+    page.getByRole("tab").getByText(anvilTabs.datasets.tabName)
+  ).toBeVisible();
+
+  // Select a filter
+  await page
+    .getByRole("button")
+    .getByText(
+      filterRegex(anvilFilters[Math.floor(Math.random() * anvilFilters.length)])
+    )
+    .click();
+  // Expect all checkboxes to be unchecked initially and to work properly
+  await expect(page.getByRole("checkbox").first()).toBeVisible();
+  const all_checkboxes = await page.getByRole("checkbox").all();
```

Comment:
Could you update this variable to be camel-case (and also any other snake-case local variables throughout)?